contrib/libb64/cdecode.c

```diff
@@ -7,16 +7,16 @@ For details, see http://sourceforge.net/projects/libb64
 
 #include "libb64/cdecode.h"
 
-int base64_decode_value(char value_in) {
-	static const char decoding[] = { 62, -1, -1, -1, 63, 52, 53, 54, 55, 56, 57, 58, 59, 60, 61, -1, -1, -1, -2, -1,
-		                             -1, -1, 0,  1,  2,  3,  4,  5,  6,  7,  8,  9,  10, 11, 12, 13, 14, 15, 16, 17,
-		                             18, 19, 20, 21, 22, 23, 24, 25, -1, -1, -1, -1, -1, -1, 26, 27, 28, 29, 30, 31,
-		                             32, 33, 34, 35, 36, 37, 38, 39, 40, 41, 42, 43, 44, 45, 46, 47, 48, 49, 50, 51 };
-	static const char decoding_size = sizeof(decoding);
+int base64_decode_value(int value_in) {
+	static const int decoding[] = { 62, -1, -1, -1, 63, 52, 53, 54, 55, 56, 57, 58, 59, 60, 61, -1, -1, -1, -2, -1,
+		                            -1, -1, 0,  1,  2,  3,  4,  5,  6,  7,  8,  9,  10, 11, 12, 13, 14, 15, 16, 17,
+		                            18, 19, 20, 21, 22, 23, 24, 25, -1, -1, -1, -1, -1, -1, 26, 27, 28, 29, 30, 31,
+		                            32, 33, 34, 35, 36, 37, 38, 39, 40, 41, 42, 43, 44, 45, 46, 47, 48, 49, 50, 51 };
+	static const int decoding_size = sizeof(decoding);
```

Comment:
I think the previous code might have been relying on `sizeof(char) == 1` here. Should this be `sizeof(decoding) / sizeof(int)`?

Reply:
Good point, though I wonder why I didn't see this in my runs of the unit test. I'm also uncertain what this is supposed to be doing when `value_in == decoding_size`.